ods_ci/tests/Tests/400__ods_dashboard/415__ods_dashboard_projects/415__ods_dashboard_projects_notebook_images.robot

```diff
@@ -41,6 +41,28 @@ Verify User Can Create A Workbench Using Intel AiKit Image
     ...    project_title=${PRJ_TITLE}    username=${TEST_USER_3.USERNAME}
     ...    password=${TEST_USER_3.PASSWORD}    auth_type=${TEST_USER_3.AUTH_TYPE}
 
+Verify User Can Create A Workbench Using Code Server Image
+    [Documentation]    Verifies that a workbench can be created using Code Server image
+    [Tags]    Sanity    Tier1    vscode
+    Set Test Variable   ${IMG_NAME}    Code Server
+    Set Test Variable   ${WORKBENCH_TITLE}    vscode
+    Set Test Variable   ${PV_NAME}    vscodepv
+    Set Test Variable   ${PV_DESCRIPTION}    PV for vscode
+    Go To RHODS Dashboard
+    Open Data Science Project Details Page       project_title=${PRJ_TITLE}
+    Create Workbench    workbench_title=${WORKBENCH_TITLE}  workbench_description=workbench for testing
+    ...        prj_title=${PRJ_TITLE}    image_name=${IMG_NAME}  version=${NONE}    deployment_size=Small
+    ...        storage=Persistent  pv_name=${PV_NAME}  pv_existent=${FALSE}
+    ...        pv_description=${PV_DESCRIPTION}  pv_size=1
+    ...        press_cancel=${FALSE}    envs=${NONE}
+    Wait Until Workbench Is Started     workbench_title=${WORKBENCH_TITLE}
+    Open Data Science Projects Home Page
+    Wait Until Project Is Listed    project_title=${PRJ_TITLE}
+    Launch And Access Workbench From Projects Home Page    workbench_title=${WORKBENCH_TITLE}
+    ...    project_title=${PRJ_TITLE}    username=${TEST_USER_3.USERNAME}
+    ...    password=${TEST_USER_3.PASSWORD}    auth_type=${TEST_USER_3.AUTH_TYPE}
+    ...    expected_ide=VSCode
+
```

Comment:
what about ensuring VS Code can be actually used? like running some code inside

Reply:
Yeah, I'll need to understand how to do it properly first. The UI written for VS Code is more JS event based compared to our RHOAI dashboard or JupyterLab. I was thinking about having at least this basic one in place and then try to do more.

Ideal situation would be to be able to actually perform such check in the upstream repository instead here, but we're not there yet...

Reply:
>Ideal situation would be to be able to actually perform such check in the upstream repository instead here, but we're not there yet...


wouldn't it still be part of an E2E test to validate it works fine?

Reply:
It depends on the actual check - if we would like to check an actual integration with the product that wasn't done in upstream repo, then yes. But if we would like to just check e.g. that we can create a file with some simple code and are able to compile it, then we don't need to do it here IMHO. We can just check the image tag that it matches what we expected/tested upstream... but we're far from this anyway...